gips/data/core.py

```diff
@@ -404,7 +414,24 @@ def discover(cls, tile, date, asset=None):
             files = utils.find_files(cls._assets[a]['pattern'], tpath)
             # more than 1 asset??
             if len(files) > 1:
-                raise Exception("Duplicate(?) assets found: {}".format(files))
+                fv = sorted(
+                    [(f, cls(f)._version) for f in files],
+                    key=lambda x: -x[1]
+                )
+                if fv[0][1] not in zip(*fv[1:])[1]:
+                   # greatest version is not duplicated
+                    # remove lesser versions
+                    utils.verbose_out('fn,                                version')
+                    for fn, ver in fv:
+                        utils.verbose_out('{},{}'.format(fn, ver))
+                    keep = fv[0][0]
+                    rm = zip(*fv[1:])[0]
+                    RemoveFiles(rm)
+                    files = keep
```

Comment:
Shouldn't this be `file = [keep]` so that the following `len` works as expected?

Reply:
yes.  I was only testing from `Asset.fetch` which uses the return as a boolean.  `Asset._archivefile` seems to actually care about what is returned.